injection/sharedmain/main.go

```diff
@@ -165,6 +166,14 @@ func toControllerConstructors(namedCtors []injection.NamedControllerConstructor)
 // MainWithContext runs the generic main flow for controllers and
 // webhooks. Use MainWithContext if you do not need to serve webhooks.
 func MainWithContext(ctx context.Context, component string, ctors ...injection.ControllerConstructor) {
+	// Allow configuration of threads per controller
+	if val, ok := os.LookupEnv("K_THREADS_PER_CONTROLLER"); ok {
+		threadsPerController, err := strconv.Atoi(val)
+		if err != nil {
+			log.Fatalf("failed to parse value %q of K_THREADS_PER_CONTROLLER: %v\n", val, err)
+		}
+		controller.DefaultThreadsPerController = threadsPerController
```

Comment:
FWIW it is a global variable so that binary entrypoints can use the following to surface it as a flag, but I'm also totally fine with having an environment variable for it as well.
```
flag.StringVar(&controller.DefaultThreadsPerController, ...)
```

Reply:
It was a baked in flag and _those kinds_ flags have caused so many footguns I asked for an env var. And yes, because it is exposed, an application can choose to use a flag if they wanted. +1

Reply:
I wasn't suggesting a flag in `package sharedmain` (yes, library = footgun), I was suggesting a flag in the consumer's `package main` (<-- binary entrypoint).